packages/shared/src/components/fields/MarkdownInput/CommentMarkdownInput.tsx

```diff
@@ -0,0 +1,209 @@
+  );
+  const { requestMethod, isCompanion } = useRequestProtocol();
+  const onSuccess = (comment: Comment) => {
+    if (!comment) return;
+
+    const comments = ['post_comments', postId];
+    client.setQueryData<PostCommentsData>(comments, (data) => {
+      const copy = { ...data };
+
+      if (!editCommentId) {
+        const edge = generateCommentEdge(comment);
+
+        if (!parentCommentId) {
+          copy.postComments.edges.push(edge);
+          return copy;
+        }
+
+        const index = copy.postComments.edges.findIndex(
+          ({ node }) => node.id === parentCommentId,
+        );
+        copy.postComments.edges[index].node.children.edges.push(edge);
+
+        return copy;
+      }
+
+      if (!parentCommentId) {
+        const index = copy.postComments.edges.findIndex(
+          ({ node }) => node.id === editCommentId,
+        );
+        copy.postComments.edges[index].node = {
+          ...comment,
+          children: copy.postComments.edges[index].node.children,
+        };
+        return copy;
+      }
+
+      const parent = copy.postComments.edges.find(
+        ({ node }) => node.id === parentCommentId,
+      );
+      const index = parent.node.children.edges.findIndex(
+        ({ node }) => node.id === editCommentId,
+      );
+      parent.node.children.edges[index].node = {
+        ...comment,
+        children: parent.node.children,
+      };
+      return copy;
+    });
+
+    if (!editCommentId) {
+      const { post } = client.getQueryData<PostData>(getPostByIdKey(postId));
+      updatePostCache(client, postId, {
+        numComments: post.numComments + 1,
+      });
+    }
+
+    if (onCommented) onCommented(comment, !editCommentId);
+  };
+
+  const mutation = parentCommentId
+    ? COMMENT_ON_COMMENT_MUTATION
+    : COMMENT_ON_POST_MUTATION;
+  const { mutateAsync: onComment, isLoading: isCommenting } = useMutation(
+    (variables: SubmitComment) =>
+      requestMethod(graphqlUrl, mutation, variables, {
+        requestKey: JSON.stringify(key),
+      }),
+    {
+      onSuccess: (data) => onSuccess(data.comment),
+    },
+  );
+
+  useBackgroundRequest(key, {
+    enabled: isCompanion,
+    callback: ({ res }) => onSuccess(res.comment),
+  });
+
+  const { mutateAsync: editComment, isLoading: isEditing } = useMutation(
+    (variables: SubmitComment) =>
+      requestMethod(graphqlUrl, EDIT_COMMENT_MUTATION, variables, {
+        requestKey: JSON.stringify(key),
+      }),
+    { onSuccess: (data) => onSuccess(data.comment) },
+  );
+
+  const onSubmit: FormEventHandler<HTMLFormElement> = (e) => {
+    e.preventDefault();
+
+    const { content } = formToJson(e.currentTarget);
+
+    if (editCommentId) {
+      return editComment({ id: editCommentId, content });
+    }
+
+    return onComment({ content, id: parentCommentId ?? postId });
+  };
+
+  useEffect(() => {
+    markdownRef?.current?.textareaRef?.current?.focus();
+  }, []);
```

Comment:
Does this need to be in a useEffect? Think you can even simply just call if in the render

Reply:
Do you mean being inside the `ref` callback on the form element?